pfcli/configurator/job.py

```diff
@@ -510,3 +522,140 @@ def get_job_request_body(self) -> Dict[str, Any]:
+                    },
+                    "required": [
+                        "name",
+                    ],
+                    "additionalProperties": False,
+                },
+                "plugin": {
+                    "type": "object",
+                    "properties": {
+                        "wandb": {
+                            "type": "object",
+                            "properties": {"credential_id": {"type": "string"}},
+                            "required": ["credential_id"],
+                        }
+                    },
+                    "additionalProperties": False,
+                },
+            },
+            "required": [
+                "job_setting",
+            ],
+            "additionalProperties": False,
+        }
+
+    def update_config(
+        self,
+        vm: Optional[str] = None,
+        num_devices: Optional[int] = None,
+        name: Optional[str] = None,
+    ) -> None:
+        """In-place update the job configuration.
+
+        Args:
+            vm (Optional[str], optional): VM name. Defaults to None.
+            num_devices (Optional[int], optional): The number of devices. Defaults to None.
+            name (Optional[str], optional): Job name. Defaults to None.
+
+        """
+        if num_devices is not None or vm is not None:
+            secho_error_and_exit("Cannot configure VM settings for the predefined job.")
+        if name is not None:
+            self._config["name"] = name
+
+    def get_job_request_body(self) -> Dict[str, Any]:
+        """Get a request body for the REST API call.
+
+        Returns:
+            Dict[str, Any]: Post-processed job request body.
+
+        """
+        body = deepcopy(self._config)
+
+        data_client: ProjectDataClientService = build_client(ServiceType.PROJECT_DATA)
+
+        if "data" in body:
+            data_name: str = body["data"]["name"]
+            if data_name.startswith("huggingface:"):
+                body["public_source"] = {
+                    "data": {
+                        "provider": "huggingface",
+                        "name": data_name.lstrip("huggingface:"),
+                    }
+                }
+                del body["data"]
+            else:
+                data_id = data_client.get_id_by_name(data_name)
+                if data_id is None:
+                    secho_error_and_exit(f"Dataset ({data_name}) is not found.")
+                body["data"]["id"] = data_id
+                del body["data"]["name"]
+
+        return body
+
+
+def get_configurator(f: IO) -> Union[CustomJobConfigurator, PredefinedJobConfigurator]:
+    try:
+        config: Dict[str, Any] = yaml.safe_load(f)
+    except yaml.YAMLError as e:
+        secho_error_and_exit(f"Error occurred while parsing config file: {e!r}")
+
+    type_to_cls = {
+        "custom": CustomJobConfigurator,
+        "predefined": PredefinedJobConfigurator,
+    }
+    cls = type_to_cls[config["job_setting"]["type"]]
```

Comment:
At this point, since we have not called `validate` yet, I think we should check whether `job_setting` exists in `config` and `type` exists in `config["job_setting"]` for a kind exit.